x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/mapper/MapperUtils.java

```diff
@@ -53,7 +58,15 @@ private MapperUtils() {}
 
     static PhysicalPlan mapLeaf(LeafPlan p) {
         if (p instanceof Row row) {
-            return new RowExec(row.source(), row.fields());
+            // return new RowExec(row.source(), row.fields());
+            // convert row into local relation
+            List<Alias> fields = row.fields();
+            List<Object> values = new ArrayList<>(fields.size());
+            for (Alias field : fields) {
+                values.add(field.child().fold());
+            }
+            Block[] blocks = BlockUtils.fromListRow(PlannerUtils.NON_BREAKING_BLOCK_FACTORY, values);
+            p = new LocalRelation(row.source(), row.output(), LocalSupplier.of(blocks));
```

Comment:
Makes sense. I guess we can purge `RowExec` entirely? (It shouldn't have ever been serialised.)
But was this intended in this PR?

Reply:
It's a workaround to have the mapping works - otherwise it fails since it expects a local relationship and the row throws it off.
It's a topic we've identified before:
https://github.com/elastic/elasticsearch/issues/104960, https://github.com/elastic/elasticsearch/issues/106653